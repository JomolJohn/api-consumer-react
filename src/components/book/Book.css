.wrapper {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(350px, 1fr));
    grid-gap: 25px;
}
.card {
    display: flex;
    flex-direction: column;
    border: 3px solid #3498db;
    text-align: center;
}
.card__content {
    flex-grow: 1;
    display: flex;
    flex-direction: column;
}
.card__link {
    margin-top: auto;
}
.card__book {
    text-align: center;
}