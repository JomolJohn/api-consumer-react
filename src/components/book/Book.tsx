import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { getBook } from '../../api/book';
import { IBookDetail } from '../../api/bookInterface';
import Loader from '../../common/Loader';

const Book = () => {

    const [isLoading, setIsLoading] = useState<boolean>(true);
    const [bookInfo, setBookInfo] = useState<IBookDetail>();

    const { bookId } = useParams();

    useEffect(() => {
        const fetchData = async () => {
            if (bookId) {
                await getBook(bookId).then(res => {
                    const bookData = res.data;
                    setBookInfo(bookData);
                }).catch(error => {
                    console.log(error);
                }).finally(() => {
                    setIsLoading(false);
                });
            }
        }
        fetchData();
    }, [bookId]);

    const renderBookItem = () => {
        return (
            <article className="card__book">
                <div className="card__content">
                    <h2>{bookInfo?.title}</h2>
                    <span>{bookInfo?.author}</span>
                    <span>{bookInfo?.year}</span>
                    <p className="card_link"><a href={bookInfo?.wiki_url } target="_blank" rel="noreferrer">Read more</a></p>
                </div>
            </article >
        );
    }

    return (
        <div>
            { isLoading ? <Loader /> : null }
            { bookInfo ? renderBookItem() : null }
        </div>
    );
};
export default Book;