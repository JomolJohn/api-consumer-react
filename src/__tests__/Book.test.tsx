import React from "react";
import { mount, shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";
import Book from '../components/book/Book';

describe('>>> Book component renders a specefic Book information', () => {

    it("should call setState with initial values on component mount", () => {
        jest.spyOn(React, "useState");
        const component = shallow(<Book />);
        expect(enzymeToJson(component)).toMatchSnapshot();
    });
});