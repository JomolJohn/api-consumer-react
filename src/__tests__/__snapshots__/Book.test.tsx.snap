// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Book component renders a specefic Book information should call setState with initial values on component mount 1`] = `
<div>
  <Loader />
</div>
`;
